src/ip_notifier.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+
+Copyright (c) 2016, Steven Siloti
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+* Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+* Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in
+the documentation and/or other materials provided with the distribution.
+* Neither the name of the author nor the names of its
+contributors may be used to endorse or promote products derived
+from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#include "libtorrent/ip_notifier.hpp"
+
+#if defined TORRENT_WINDOWS && !defined TORRENT_BUILD_SIMULATOR
+#include <iphlpapi.h>
+#endif
+
+using namespace std::placeholders;
+
+namespace libtorrent
+{
+	ip_change_notifier::ip_change_notifier(io_service& ios)
+#if defined TORRENT_BUILD_SIMULATOR
+#elif TORRENT_USE_NETLINK
+		: m_socket(ios
+			, netlink::endpoint(netlink(NETLINK_ROUTE), RTMGRP_IPV4_IFADDR | RTMGRP_IPV6_IFADDR))
+#elif defined TORRENT_WINDOWS
+		: m_hnd(ios, WSACreateEvent())
+#endif
+	{
+#if defined TORRENT_BUILD_SIMULATOR
+		TORRENT_UNUSED(ios);
+#elif defined TORRENT_WINDOWS
+		m_ovl.hEvent = m_hnd.native_handle();
+#elif !TORRENT_USE_NETLINK
+		TORRENT_UNUSED(ios);
+#endif
+	}
+
+	ip_change_notifier::~ip_change_notifier()
+	{
+#if defined TORRENT_WINDOWS && !defined TORRENT_BUILD_SIMULATOR
+		cancel();
+		m_hnd.close();
```

Comment:
I thought the same thing, but it turns out `object_handle` doesn't close the handle in its destructor.